params/precompile_config_test.go

```diff
@@ -57,30 +59,132 @@ func TestValidateWithChainConfig(t *testing.T) {
+				},
+				{
+					TxAllowListConfig: precompile.NewDisableTxAllowListConfig(big.NewInt(2)),
+				},
+				{
+					TxAllowListConfig: &precompile.TxAllowListConfig{
+						AllowListConfig:   precompile.AllowListConfig{AllowListAdmins: admins, EnabledAddresses: admins},
+						UpgradeableConfig: precompile.UpgradeableConfig{BlockTimestamp: big.NewInt(3)},
+					},
+				},
+			},
+			expectedError: "cannot set address",
+		},
+		{
+			name: "invalid initial fee manager config",
+			upgrades: []PrecompileUpgrade{
+				{
+					FeeManagerConfig: &precompile.FeeConfigManagerConfig{
+						AllowListConfig:   precompile.AllowListConfig{AllowListAdmins: admins},
+						UpgradeableConfig: precompile.UpgradeableConfig{BlockTimestamp: big.NewInt(3)},
+						InitialFeeConfig: &commontype.FeeConfig{
+							GasLimit: big.NewInt(-1),
+						},
+					},
+				},
+			},
+			expectedError: "gasLimit = -1 cannot be less than or equal to 0",
 		},
 	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			require := require.New(t)
+			baseConfig := *SubnetEVMDefaultChainConfig
+			config := &baseConfig
+			config.PrecompileUpgrades = tt.upgrades
+
+			err := config.Verify()
+			if tt.expectedError == "" {
+				require.NoError(err)
+			} else {
+				require.ErrorContains(err, tt.expectedError)
+			}
+		})
+	}
+}
 
-	// check this config is valid
-	err := config.VerifyPrecompileUpgrades()
-	assert.NoError(t, err)
+func TestVerifyPrecompiles(t *testing.T) {
+	admins := []common.Address{{1}}
+	tests := []struct {
+		name          string
+		upgrade       PrecompileUpgrade
+		expectedError string
+	}{
+		{
+			name: "invalid allow list config in tx allowlist",
+			upgrade: PrecompileUpgrade{
+				TxAllowListConfig: &precompile.TxAllowListConfig{
+					AllowListConfig:   precompile.AllowListConfig{AllowListAdmins: admins, EnabledAddresses: admins},
+					UpgradeableConfig: precompile.UpgradeableConfig{BlockTimestamp: big.NewInt(3)},
+				},
+			},
+			expectedError: "cannot set address",
+		},
+		{
+			name: "invalid initial fee manager config",
+			upgrade: PrecompileUpgrade{
+				FeeManagerConfig: &precompile.FeeConfigManagerConfig{
+					AllowListConfig:   precompile.AllowListConfig{AllowListAdmins: admins},
+					UpgradeableConfig: precompile.UpgradeableConfig{BlockTimestamp: big.NewInt(3)},
+					InitialFeeConfig: &commontype.FeeConfig{
+						GasLimit: big.NewInt(-1),
+					},
+				},
+			},
+			expectedError: "gasLimit = -1 cannot be less than or equal to 0",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			require := require.New(t)
+			baseConfig := *SubnetEVMDefaultChainConfig
+			config := &baseConfig
+			config.PrecompileUpgrade = tt.upgrade
+
+			err := config.Verify()
+			if tt.expectedError == "" {
+				require.NoError(err)
+			} else {
+				require.ErrorContains(err, tt.expectedError)
+			}
+		})
+	}
 }
 
 func TestValidateRequiresSortedTimestamps(t *testing.T) {
```

Comment:
same idea here re. renaming `Validate` to `Verify`

Reply:
done